aesd-char-driver/main.c

```diff
@@ -27,6 +28,141 @@ MODULE_LICENSE("Dual BSD/GPL");
 
 struct aesd_dev aesd_device;
 
+
+
+static size_t aesd_circ_count(const struct aesd_circular_buffer *cb)
+{
+    if (cb->full) 
+    {
+    	return AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED;
+    }
+    if (cb->in_offs >= cb->out_offs)
+    {
+    	 return cb->in_offs - cb->out_offs;
+    }
+    return AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED - cb->out_offs + cb->in_offs;
+}
+
+
+static size_t aesd_bytes_before_idx(const struct aesd_circular_buffer *cb, uint8_t idx)
```

Comment:
I like the idea to add helper functions.  These might be good to place in the circular buffer file itself.